asreview/webapp/app.py

```diff
@@ -101,11 +103,13 @@ def load_user(user_id):
         with app.app_context():
             # create tables in case they don't exist
             DB.create_all()
-
         # store oauth config in oauth handler
         if bool(app.config.get("OAUTH", False)):
             app.config["OAUTH"] = OAuthHandler(app.config["OAUTH"])
 
+        if bool(app.config.get("REMOTE_USER", False)):
```

Comment:
You can configure both `REMOTE_USER` and `OAUTH`. Does it ever make sense to have both configured? If not, maybe it makes sense not to allow configuring both, then there is less of a chance of mistakes happening.

Reply:
I was also wondering about this. In principle, yes, you could allow multiple authentication methods on a single instance of the app. For instance, the webserver could set or not set `REMOTE_USER` depending on the IP address of the client, so that for a certain (say, internal) IP range remote user auth is allowed, but for different IP ranges, users still need to login using regular DB auth or OAuth.

However, I do think this use case will be rare, and the benefit of disallowing remote user auth if any other form of authentication is enabled is that it reduces the chance of misconfiguration.

For what it's worth, Flask-Appbuilder [forces](https://flask-appbuilder.readthedocs.io/en/v3.2.0/security.html#authentication-methods) the user to choose a single authentication type. Perhaps we should go with that approach, and change when it turns out there is demand for mixing auth methods?

Reply:
On the other hand: at present there is no configuration setting that enables/disables ordinary password authentication. So remote auth and password auth will always technically co-exist. Maybe then it makes sense for both of these to co-exist with OAuth, as well?

Alternatively, I think a bigger refactor should be implemented that requires the user to set, exclusively, either password authentication, OAuth authentication, or remote user authentication.

Reply:
That use case for multiple authentication methods sounds a bit exotic, and I do think there is value in allowing only a single authentication method. Besides misconfiguration, the code should also be a bit easier to understand if you can have only a single authentication method enabled.

But there's no need to do that right away. I'll make an issue so we don't forget and then you can just continue without too much trouble with this PR.